core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -190,11 +211,11 @@ class BrokerToControllerRequestThread(
       if (currentTimeMs - request.createdTimeMs >= retryTimeoutMs) {
         requestIter.remove()
         request.callback.onTimeout()
-      } else if (activeController.isDefined) {
+      } else if (activeControllerAddress().isDefined) {
         requestIter.remove()
         return Some(RequestAndCompletionHandler(
           time.milliseconds(),
-          activeController.get,
+          activeControllerAddress().get,
```

Comment:
The usage is a tad suspicious because the atomic reference suggests that the value could change. I guess we are ok because the value will only be overwritten in the same thread that is calling `generateRequests`, but it might be worth rewriting this part anyway. For example:
```scala
} else {
  val controllerAddress = activeControllerAddress()
  if (controllerAddress.isDefined) {
...
```